---
title: C6320
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
author: mikeblome
ms.author: mblome
manager: wpickett
ms.workload:
- multiple
ms.openlocfilehash: fef29c7f6e341cf4cd31991b24bfc209bd902b15
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 04/23/2019
ms.locfileid: "62781470"
---
# <a name="c6320"></a>C6320
Warnung C6320: Der Ausnahmefilterausdruck ist die EXCEPTION_EXECUTE_HANDLER-Konstante. Hierdurch werden möglicherweise Ausnahmen maskiert, für die keine Behandlung vorgesehen war.

 Diese Warnung gibt an, den Nebeneffekt, dass der Verwendung von EXCEPTION_EXECUTE_HANDLER-Konstante in __except Block. In diesem Fall die Anweisung in der \__except-Block wird immer ausgeführt, um die Ausnahme, einschließlich Ausnahmen, die Sie nicht in einer bestimmten Funktion behandeln möchten. Es wird empfohlen, dass Sie die Ausnahme vor der Behandlung zu überprüfen.

## <a name="example"></a>Beispiel
 Der folgende Code generiert diese Warnung, weil der __except-block versuchen, die Ausnahmen aller Art zu behandeln:

```
#include <stdio.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;  // might cause access violation exception
      // code ...
   }
   __except(EXCEPTION_EXECUTE_HANDLER) // warning
   {
      puts("in except");
      // code ...
   }
}
```

 Verwenden Sie zum Beheben dieser Warnung `GetExceptionCode` zu prüfen, eine bestimmte Ausnahme vor der Behandlung wie im folgenden Code gezeigt:

```
#include <stdio.h>
#include <windows.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;    // might cause access violation exception
      // code ...
   }
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
   {
      puts("in except");
      // code ...
   }
}
```

## <a name="see-also"></a>Siehe auch
 [try-except-Anweisung](/cpp/cpp/try-except-statement)